src/main/java/seedu/address/model/pet/comparator/PetPickUpTimeComparator.java

```diff
@@ -0,0 +1,64 @@
+package seedu.address.model.pet.comparator;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.Comparator;
+
+import seedu.address.model.attendance.AttendanceEntry;
+import seedu.address.model.attendance.PresentAttendanceEntry;
+import seedu.address.model.pet.Pet;
+
+/**
+ * Comparator class to compare pets based on pet's Pick Up Time.
+ */
+public class PetPickUpTimeComparator implements Comparator<Pet> {
+    private LocalDate today = LocalDate.now();
+
+    /**
+     * Compares pet objects based on their pick up time attribute.
+     * @param first Pet to be compared with.
+     * @param second Pet to be compared with.
+     * @return Value signifying in the difference between the comparing attribute.
+     */
+    @Override
+    public int compare(Pet first, Pet second) {
+        AttendanceEntry firstAttendanceEntry = first.getAttendanceHashMap().getAttendanceEntry(today);
+        AttendanceEntry secondAttendanceEntry = second.getAttendanceHashMap().getAttendanceEntry(today);
+
+        boolean isFirstPresent = firstAttendanceEntry instanceof PresentAttendanceEntry;
+        boolean isSecondPresent = secondAttendanceEntry instanceof PresentAttendanceEntry;
+
+        if (isFirstPresent && isSecondPresent) {
+            return comparePresentAttendanceEntries(firstAttendanceEntry, secondAttendanceEntry);
+        } else if (isFirstPresent && !isSecondPresent) {
+            return -1;
+        } else if (!isFirstPresent && isSecondPresent) {
+            return 1;
+        } else {
+            return 0;
+        }
+    }
```

Comment:
Maybe can consider adding some tests for these methods. Just in case they are prone to bugs or some unexpected input that user may give. Same for other comparator classes.

Reply:
Added tests for all 5 fields!